web/documentserver-example/csharp/FormatManager.cs

```diff
@@ -0,0 +1,188 @@
+                .ToList();
+        }
+
+        public static List<string> ConvertibleExtensions()
+        {
+            return Convertible()
+                .Select(format => format.Extension())
+                .ToList();
+        }
+
+        public static List<Format> Convertible()
+        {
+            return All()
+                .Where(format => (format.Type == "cell" && format.Convert.Contains("xlsx"))
+                                || (format.Type == "slide" && format.Convert.Contains("pptx"))
+                                || (format.Type == "word" && format.Convert.Contains("docx")))
+                .ToList();
+        }
+
+        public static List<string> SpreadsheetExtensions()
+        {
+            return Spreadsheets()
+                .Select(format => format.Extension())
+                .ToList();
+        }
+
+        public static List<Format> Spreadsheets()
+        {
+            return All()
+                .Where(format => format.Type == "cell")
+                .ToList();
+        }
+
+        public static List<string> PresentationExtensions()
+        {
+            return Presentations()
+                .Select(format => format.Extension())
+                .ToList();
+        }
+
+        public static List<Format> Presentations()
+        {
+            return All()
+                .Where(format => format.Type == "slide")
+                .ToList();
+        }
+
+        public static List<string> DocumentExtensions()
+        {
+            return Documents()
+                .Select(format => format.Extension())
+                .ToList();
+        }
+
+        public static List<Format> Documents()
+        {
+            return All()
+                .Where(format => format.Type == "word")
+                .ToList();
+        }
+
+        public static List<string> AllExtensions()
+        {
+            return All()
+                .Select(format => format.Extension())
+                .ToList();
+        }
+
+        public static List<Format> All()
+        {
+            var path = GetPath();
+            var lines = File.ReadLines(path, Encoding.UTF8);
+            var contents = string.Join(Environment.NewLine, lines);
+            var formats = JsonConvert.DeserializeObject<Format[]>(contents);
+            return formats.ToList();
+        }
+
+        private static string GetPath()
+        {
+            string path = Path.Combine(GetDirectory(), "onlyoffice-docs-formats.json");
+            if (File.Exists(path))
+            {
+                return path;
+            }
+            else
+            {
+                throw new FileNotFoundException("The JSON file does not exist.");
+            }
+        }
+
+        private static string GetDirectory()
+        {
+            string directory = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "assets", "document-formats");
+            return Path.GetFullPath(directory);
+        }
+    }
```

Comment:
Do you really have to use 2 functions? Like for example `FillableExtensions()` and `Fillable()`.

Reply:
> Do you really have to use 2 functions in FormatManager? Like for example `FillableExtensions()` and `Fillable()`.

Actually, there's no good reason. I looked at how it was implemented in other examples and did the same. If we need an array of action data in the future, we have a ready method.